### PR TITLE
Feature/card component

### PR DESCRIPTION
**Pull request by jlu255**

### CODE REVIEW

File: src/components/Card/Card.tsx [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  buttonTitle?: string;
+}
+// export interface ImgHTMLAttributes<T> {
+//   src?: string;
+// }
+
+export type patCardProps = ICardProps;
+/**
+ * A card with a button to show more
+ *
```
[the remark this answers is not in the document]
Comment (jlu255) [reply src/components/Card/Card.tsx:40]: sure! done

File: src/components/Card/Card.tsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        <img src={props.cardImgSrc} data-testid='image-element' />
+      </div>
+      <div className={styleClasses + ' body'} data-testid='card-body-element'>
+        <h5>{props.cardTitle}</h5>
+        <p>{props.cardParagraph}</p>
+        <Button
+          onClick={props.btnOnClick}
+          data-testid='button-element'
+          className='btn btn-primary'
+          href={props.buttonHref}
```
[the remark this answers is not in the document]
Comment (jlu255) [reply src/components/Card/Card.tsx]: sure!

File: src/components/Card/Card.tsx [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+      <div className={styleClasses + ' image'} data-testid='card-image-element'>
+        <img src={props.cardImgSrc} data-testid='image-element' />
+      </div>
+      <div className={styleClasses + ' body'} data-testid='card-body-element'>
+        <h5>{props.cardTitle}</h5>
+        <p>{props.cardParagraph}</p>
+        <Button
+          onClick={props.btnOnClick}
+          data-testid='button-element'
+          className='btn btn-primary'
```
[the remark this answers is not in the document]
Comment (jlu255) [reply src/components/Card/Card.tsx]: sure, already fixed that

File: src/components/Card/Card.tsx [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+  /** set card content */
+  cardParagraph?: string;
+  /** set card image source */
+  cardImgSrc?: string;
+  /** set action on bottun clicked */
+  btnOnClick?: () => void;
+  /** set the redirect page on button clicked*/
+  buttonHref?: string;
+  /** set button title */
+  buttonTitle?: string;
```
[the remark this answers is not in the document]
Comment (jlu255) [reply src/components/Card/Card.tsx:30]: sure, already fixed that

File: src/components/Card/Card.tsx [hunk 5]
```diff
@@ -0,0 +1,89 @@
+import React, { FC } from 'react';
+import Button from '../Button/Button';
+import { classNames } from '../../utils/classNames';
+
+export type CardSize = 'lg' | 'sm';
+export type CardMode = 'default' | 'horizontal';
+export type CardTheme = 'dark' | 'purple' | 'blue' | 'yellow';
```
[the remark this answers is not in the document]
Comment (jlu255) [reply src/components/Card/Card.tsx]: sure! I have added 'primary' to theme

File: src/components/Card/Card.tsx [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+  /** set card type */
+  cardType?: CardType;
+  /** set card theme */
+  cardTheme?: CardTheme;
+  /** set card size */
+  cardSize?: CardSize;
+  /** set card title */
+  cardTitle?: string;
+  /** set card content */
+  cardParagraph?: string;
```
[the remark this answers is not in the document]
Comment (jlu255) [reply src/components/Card/Card.tsx:24]: should i put 'any' for the type of cardParagraph in that case?

[the remark this answers is not in the document]
Comment (YuhanLin1105) [reply src/components/Card/Card.tsx:24]: You could use React.ReactNode.

[the remark this answers is not in the document]
Comment (jlu255) [reply src/components/Card/Card.tsx]: Sure! fix that

**Pull request merged**